File: pegasusX/apps/payload-app-android/app/src/main/java/com/pegasus/payload/ui/home/TruckSidebar.kt
package com.pegasus.payload.ui.home

import androidx.compose.ui.res.stringResource

import androidx.compose.animation.animateColorAsState
import com.pegasus.payload.ui.components.PayloadSectionTitle
import com.pegasus.payload.ui.components.ExplainStatusBanner
import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.spring
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.GridItemSpan
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.LocalShipping
import androidx.compose.material.icons.filled.Menu
import androidx.compose.material.icons.filled.MenuOpen
import androidx.compose.material3.Button
import androidx.compose.material3.ElevatedCard
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.pegasus.payload.data.model.Truck
import com.pegasus.payload.data.model.SealCompletedManifestResult
import com.pegasus.design.PegasusStatePane
import com.pegasus.design.PegasusStateKind
import com.pegasus.design.PegasusLoadingState
import com.pegasus.payload.R

@Composable
fun TruckListPane(
    trucks: List<Truck>,
    selectedTruckId: String?,
    loading: Boolean,
    error: String?,
    batchReadyCount: Int,
    batchSealing: Boolean,
    batchSealFailures: List<SealCompletedManifestResult> = emptyList(),
    isExpanded: Boolean,
    onToggleExpanded: () -> Unit,
    onFinalizeBatch: () -> Unit,
    onSelect: (String) -> Unit,
) {
    Surface(
        color = MaterialTheme.colorScheme.surface,
        modifier = Modifier
            .width(if (isExpanded) 320.dp else 72.dp)
            .fillMaxHeight(),
    ) {
        Column(Modifier.fillMaxSize()) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 12.dp, vertical = 12.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                if (isExpanded) {
                    PayloadSectionTitle(
                        title = "Manifest board",
                        subtitle = "DRAFT · LOADING · SEALED · DISPATCHED",
                        modifier = Modifier.weight(1f),
                    )
                }
                IconButton(onClick = onToggleExpanded) {
                    Icon(
                        if (isExpanded) Icons.Default.MenuOpen else Icons.Default.Menu,
                        contentDescription = if (isExpanded) "Collapse truck list" else "Expand truck list",
                    )
                }
            }
            if (!isExpanded) {
                LazyVerticalGrid(
                    columns = GridCells.Adaptive(minSize = 340.dp),
                    contentPadding = PaddingValues(vertical = 8.dp),
                ) {
                    items(trucks, key = { it.id }) { truck ->
                        val selected = truck.id == selectedTruckId
                        IconButton(onClick = { onSelect(truck.id) }) {
                            Icon(
                                Icons.Default.LocalShipping,
                                contentDescription = truck.label.ifBlank { truck.licensePlate },
                                tint = if (selected) {
                                    MaterialTheme.colorScheme.onPrimaryContainer
                                } else {
                                    MaterialTheme.colorScheme.onSurfaceVariant
                                },
                            )
                        }
                    }
                }
                return@Column
            }
            if (loading) LinearProgressIndicator(Modifier.fillMaxWidth())
            if (batchReadyCount > 1) {
                ElevatedCard(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 12.dp, vertical = 8.dp),
                ) {
                    Column(Modifier.padding(12.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                        Text(
                            stringResource(R.string.mobile_payload_ui_batchreadycount_trucks_ready_to_finalize, batchReadyCount),
                            style = MaterialTheme.typography.titleSmall,
                        )
                        Button(
                            onClick = onFinalizeBatch,
                            enabled = !batchSealing,
                            modifier = Modifier.fillMaxWidth(),
                        ) {
                            Text(if (batchSealing) "Finalizing…" else "Seal all trucks")
                        }
                        batchSealFailures.forEach { row ->
                            Column(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .padding(top = 4.dp),
                            ) {
                                Text(stringResource(R.string.mobile_payload_ui_manifestid_status, row.manifestId, row.status),
                                    style = MaterialTheme.typography.labelMedium,
                                )
                                row.explain?.let { explain ->
                                    ExplainStatusBanner(
                                        explain = explain,
                                        fallbackTitle = row.status,
                                        modifier = Modifier.padding(top = 4.dp),
                                    )
                                }
                            }
                        }
                    }
                }
            }
            if (error != null) {
                Text(
                    error,
                    color = MaterialTheme.colorScheme.error,
                    style = MaterialTheme.typography.bodySmall,
                    modifier = Modifier.padding(horizontal = 20.dp, vertical = 8.dp),
                )
            }
            if (loading && trucks.isEmpty() && error == null) {
                PegasusLoadingState(
                    title = stringResource(R.string.mobile_payload_ui_loading_vehicles),
                    body = "Refreshing supplier fleet availability for this shift.",
                )
            } else if (!loading && trucks.isEmpty() && error == null) {
                PegasusStatePane(
                    kind = PegasusStateKind.Empty,
                    headline = "No vehicles available",
                    body = "Pull to refresh once dispatch assigns trucks.",
                )
            }
            val columns = ManifestBoard.group(trucks)
            val unassigned = ManifestBoard.unassigned(trucks)
            LazyVerticalGrid(
                columns = GridCells.Adaptive(minSize = 340.dp),
                contentPadding = PaddingValues(horizontal = 12.dp, vertical = 4.dp)
            ) {
                columns.forEach { column ->
                    item(key = "col-${column.state}", span = { GridItemSpan(maxLineSpan) }) {
                        Text(
                            text = "${column.state} · ${if (column.trucks.isEmpty()) "empty" else column.trucks.size.toString()}",
                            style = MaterialTheme.typography.labelMedium,
                            fontWeight = FontWeight.Bold,
                            modifier = Modifier.padding(top = 10.dp, bottom = 4.dp),
                        )
                    }
                    if (column.trucks.isEmpty()) {
                        item(key = "empty-${column.state}", span = { GridItemSpan(maxLineSpan) }) {
                            Text(
                                "No ${column.state.lowercase()} manifests",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant,
                                modifier = Modifier.padding(bottom = 6.dp),
                            )
                        }
                    } else {
                        items(column.trucks, key = { "${column.state}-${it.id}" }) { truck ->
                            TruckRow(truck, selected = truck.id == selectedTruckId, onClick = { onSelect(truck.id) })
                            Spacer(Modifier.height(6.dp))
                        }
                    }
                }
                if (unassigned.isNotEmpty()) {
                    item(key = "unassigned-head", span = { GridItemSpan(maxLineSpan) }) {
                        Text(
                            "No open manifest",
                            style = MaterialTheme.typography.labelMedium,
                            modifier = Modifier.padding(top = 10.dp, bottom = 4.dp),
                        )
                    }
                    items(unassigned, key = { "none-${it.id}" }) { truck ->
                        TruckRow(truck, selected = truck.id == selectedTruckId, onClick = { onSelect(truck.id) })
                    }
                }
            }
        }
    }
}

@Composable
fun TruckRow(truck: Truck, selected: Boolean, onClick: () -> Unit) {
    val bgColor by animateColorAsState(
        targetValue = if (selected) MaterialTheme.colorScheme.primaryContainer else MaterialTheme.colorScheme.surfaceContainerHigh,
        animationSpec = spring(dampingRatio = Spring.DampingRatioMediumBouncy, stiffness = Spring.StiffnessMedium),
        label = "truck-bg",
    )
    val fgColor by animateColorAsState(
        targetValue = if (selected) MaterialTheme.colorScheme.onPrimaryContainer else MaterialTheme.colorScheme.onSurface,
        label = "truck-fg",
    )
    Surface(
        color = bgColor,
        contentColor = fgColor,
        shape = RoundedCornerShape(14.dp),
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .clickable(onClick = onClick),
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier.padding(14.dp),
        ) {
            Icon(Icons.Filled.LocalShipping, contentDescription = null)
            Spacer(Modifier.size(12.dp))
            Column(Modifier.fillMaxWidth()) {
                Text(
                    truck.label.ifBlank { truck.licensePlate.ifBlank { truck.id.take(8) } },
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Medium,
                )
                Text(
                    listOfNotNull(
                        truck.licensePlate.takeIf { it.isNotBlank() },
                        truck.vehicleClass.takeIf { it.isNotBlank() },
                        ManifestBoard.canonicalState(truck.truckStatus).takeIf { it.isNotEmpty() },
                        if (truck.maxVolumeVu > 0) "${truck.usedVolumeVu}/${truck.maxVolumeVu} VU" else null,
                    ).joinToString(" • "),
                    style = MaterialTheme.typography.bodySmall,
                )
            }
        }
    }
}

File: pegasusX/apps/payload-app-ios/payload-app-ios/Views/TruckSidebar.swift
import SwiftUI

struct TruckSidebar: View {
    @Bindable var viewModel: HomeViewModel
    @Binding var isExpanded: Bool

    var body: some View {
        VStack(spacing: 0) {
            HStack {
                if isExpanded {
                    VStack(alignment: .leading, spacing: 2) {
                        Text("MANIFEST BOARD")
                            .font(.system(size: 13, weight: .bold, design: .monospaced))
                            .foregroundStyle(TermTheme.secondary)
                        Text("DRAFT · LOADING · SEALED · DISPATCHED")
                            .font(.system(size: 9, weight: .bold, design: .monospaced))
                            .foregroundStyle(TermTheme.tertiary)
                    }
                }
                Spacer(minLength: 0)
                Button {
                    withAnimation(.spring(response: 0.35, dampingFraction: 0.8)) {
                        isExpanded.toggle()
                    }
                } label: {
                    Image(systemName: "sidebar.left")
                        .foregroundStyle(TermTheme.secondary)
                }
                .buttonStyle(.plain)
            }
            .padding(.horizontal, isExpanded ? 16 : 12)
            .padding(.top, 8)

            Group {
                if viewModel.loadingTrucks && viewModel.trucks.isEmpty {
                    PayloadLoadingView(
                        title: "LOADING_VEHICLES",
                        message: "Refreshing supplier fleet availability for this shift."
                    )
                } else if viewModel.trucks.isEmpty {
                    PayloadStateView(
                        variant: .truck,
                        title: "NO_VEHICLES",
                        message: "Pull to refresh once dispatch assigns trucks.",
                        tone: .warning
                    )
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                } else if isExpanded {
                    expandedTruckList
                } else {
                    collapsedTruckRail
                }
            }
            if let err = viewModel.error, isExpanded {
                Text(err)
                    .font(.footnote)
                    .foregroundStyle(.red)
                    .padding(.horizontal)
            }
        }
    }

    private var expandedTruckList: some View {
        Group {
            if viewModel.batchReadyManifestIds.count > 1 {
                Section {
                    VStack(alignment: .leading, spacing: TermTheme.s8) {
                        PayloadSectionHeader(
                            title: "BATCH FINALIZE",
                            subtitle: "\(viewModel.batchReadyManifestIds.count) trucks ready to seal"
                        )
                        Button(viewModel.batchSealing ? "Finalizing…" : "Seal all trucks") {
                            Task { await viewModel.finalizeBatchSeal() }
                        }
                        .buttonStyle(.borderedProminent)
                        .disabled(viewModel.batchSealing)
                        ForEach(viewModel.batchSealFailures.indices, id: \.self) { idx in
                            let row = viewModel.batchSealFailures[idx]
                            VStack(alignment: .leading, spacing: 4) {
                                Text("\(row.manifestId ?? "manifest"): \(row.status ?? "failed")")
                                    .font(.caption.monospaced())
                                if let explain = row.explain {
                                    ExplainStatusBanner(explain: explain)
                                }
                            }
                        }
                    }
                    .padding(.vertical, 4)
                }
            }
            List(selection: Binding(
                get: { viewModel.selectedTruckId },
                set: { id in if let id { Task { await viewModel.selectTruck(id) } } }
            )) {
                ForEach(ManifestBoard.group(viewModel.trucks), id: \.state) { column in
                    Section(column.trucks.isEmpty ? "\(column.state) · empty" : "\(column.state) · \(column.trucks.count)") {
                        if column.trucks.isEmpty {
                            Text("No \(column.state.lowercased()) manifests")
                                .font(.caption)
                                .foregroundStyle(TermTheme.tertiary)
                        } else {
                            ForEach(column.trucks) { truck in
                                TruckRow(truck: truck)
                                    .tag(truck.id)
                            }
                        }
                    }
                }
                if !ManifestBoard.unassigned(viewModel.trucks).isEmpty {
                    Section("NO OPEN MANIFEST") {
                        ForEach(ManifestBoard.unassigned(viewModel.trucks)) { truck in
                            TruckRow(truck: truck)
                                .tag(truck.id)
                        }
                    }
                }
            }
            .refreshable { await viewModel.refreshTrucks() }
        }
    }

    private var collapsedTruckRail: some View {
        ScrollView(showsIndicators: false) {
            VStack(spacing: 8) {
                ForEach(viewModel.trucks) { truck in
                    let selected = viewModel.selectedTruckId == truck.id
                    Button {
                        Task { await viewModel.selectTruck(truck.id) }
                    } label: {
                        Image(systemName: "truck.box.fill")
                            .font(.system(size: 20, weight: selected ? .bold : .regular))
                            .foregroundStyle(selected ? TermTheme.accent : TermTheme.secondary)
                            .frame(width: 44, height: 44)
                            .background(
                                RoundedRectangle(cornerRadius: 10, style: .continuous)
                                    .fill(selected ? TermTheme.accent.opacity(0.12) : Color.clear)
                            )
                    }
                    .buttonStyle(.plain)
                }
            }
            .padding(.vertical, 8)
        }
    }
}

struct TruckRow: View {
    let truck: Truck

    var body: some View {
        HStack(spacing: 16) {
            ZStack {
                RoundedRectangle(cornerRadius: 12, style: .continuous)
                    .fill(TermTheme.accent.opacity(0.06))
                    .frame(width: 48, height: 48)
                
                Image(systemName: "truck.box.fill")
                    .font(.system(size: 20, weight: .bold))
                    .foregroundStyle(TermTheme.accent)
            }

            VStack(alignment: .leading, spacing: 4) {
                Text(displayLabel.uppercased())
                    .font(.system(size: 16, weight: .black, design: .monospaced))
                    .foregroundStyle(TermTheme.accent)
                
                Text(subtitle.uppercased())
                    .font(.system(size: 11, weight: .bold, design: .monospaced))
                    .foregroundStyle(TermTheme.secondary)
            }
        }
        .padding(.vertical, 8)
    }

    private var displayLabel: String {
        if let l = truck.label, !l.isEmpty { return l }
        if let p = truck.licensePlate, !p.isEmpty { return p }
        return "TRK-\(truck.id.prefix(6))"
    }

    private var subtitle: String {
        var parts = [truck.licensePlate, truck.vehicleClass]
            .compactMap { $0?.isEmpty == false ? $0 : nil }
        if let status = truck.truckStatus, ManifestBoard.isBoardState(status) {
            parts.append(status)
        }
        if let max = truck.maxVolumeVu, max > 0 {
            parts.append(String(format: "%.0f/%.0f VU", truck.usedVolumeVu ?? 0, max))
        }
        return parts.joined(separator: " — ")
    }
}
